### PR TITLE
Fixes that enable basic single sample overfitting

### PR DESCRIPTION
**Pull request by moritzhauschulz**
## Description

Fixes to `target_and_aux_diffusion.py` and `loss_module_latent_diffusion.py` which should enable to train the model purely with latent loss on single samples.

Did some basic lifting, but some long line issues are not related to my change so leaving out for clarity.

Test run with:
```uv run --offline  train_continue --from-run-id t0bdz7qn --config config/config_diffusion.yml --options training_config.losses.latent_diff.weight=1 training_config.losses.physical.weight=0```

Experiments are in progress and will be documented here:
https://gitlab.jsc.fz-juelich.de/hedgedoc/jaS9ZSMBQBmDefBX_ciqBA?both

## Issue Number

Is this PR a draft? Mark it as draft.

## Checklist before asking for review

-   [x] I have performed a self-review of my code
-   [ ] My changes comply with basic sanity checks:
      - I have fixed formatting issues with `./scripts/actions.sh lint`
      - I have run unit tests with `./scripts/actions.sh unit-test`
      - I have documented my code and I have updated the docstrings.
      - I have added unit tests, if relevant
-   [x] I have tried my changes with data and code:
      - I have run the integration tests with `./scripts/actions.sh integration-test`
      - (bigger changes) I have run a full training and I have written in the comment the run_id(s): `launch-slurm.py --time 60`
      - (bigger changes and experiments) I have shared a hegdedoc in the github issue with all the configurations and runs for this experiments
-   [x] I have informed and aligned with people impacted by my change:
    - for config changes: the MatterMost channels and/or a design doc
    - for changes of dependencies: the MatterMost software development channel

### CODE REVIEW

File: src/weathergen/train/loss_modules/loss_module_latent_diffusion.py [hunk 1]
```diff
@@ -92,10 +92,12 @@ def compute_loss(self, preds: dict, targets: dict, **kwargs) -> LossValues:
             for _, _, loss_fct_name in self.loss_fcts
         }
 
-        pred_tokens_all = [pl["latent_state"].patch_tokens for pl in preds.latent if pl]
+        pred_tokens_all = [pl["latent_state"].z_pre_norm for pl in preds.latent if pl]
```
Comment (clessig) [inline src/weathergen/train/loss_modules/loss_module_latent_diffusion.py:95]: Commenting out `reset()` in target_aux_calculator and this are the crucial changes?

File: src/weathergen/train/target_and_aux_diffusion.py [hunk 2]
```diff
@@ -15,11 +16,14 @@ def __init__(self, encoder, is_model_sharded=True):
         # Todo: make sure this is a frozen clone or forward without gradients in compute()
         self.encoder = encoder
 
+        apply_fct_to_blocks(self.encoder, ".*", freeze_weights)
+        apply_fct_to_blocks(self.encoder, ".*", set_to_eval)
```
Comment (moritzhauschulz) [inline src/weathergen/train/target_and_aux_diffusion.py:20]: These are also very crucial actually (freezing more than eval) @clessig

File: src/weathergen/model/model.py [hunk 3]
```diff
@@ -651,7 +651,7 @@ def forward(self, model_params: ModelParams, batch: ModelBatch) -> ModelOutput:
         t_mean = tokens.mean()
         t_std = tokens.std()
         tokens = (tokens - t_mean) / (t_std + 1e-6)
-        tokens = torch.clamp(tokens, -5.0, 5.0)
+        tokens = torch.clamp(tokens, -100.0, 100.0)
```
Comment (moritzhauschulz) [inline src/weathergen/model/model.py:654]: This is also somewhat crucial to avoid loss of information – without this you won't get perfect identity when setting forecast engine to identity. But I think we might remove this clamping eventually anyways. @clessig

**Review by moritzhauschulz — COMMENTED** [review]
The below are also important changes, particularly the one freezing the encoder.
